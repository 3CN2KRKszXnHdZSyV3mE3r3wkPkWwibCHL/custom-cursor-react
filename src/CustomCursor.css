.cursor {
  pointer-events: none;
}

.cursor-circle {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1000;
  opacity: 0;
}