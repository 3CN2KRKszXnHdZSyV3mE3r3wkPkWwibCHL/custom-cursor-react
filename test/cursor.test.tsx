import * as React from "react";
import * as ReactDOM from "react-dom";
import CustomCursor from "../src";
import { render } from "@testing-library/react";

// const DEFAULT_PROPS = {
// 	className: "",
// 	width: 50,
// 	height: 50,
// 	fill: "#000",
// 	stroke: "#000",
// 	strokeWidth: 0,
// };

describe("CustomCursor Component", () => {
	it("should render without crashing", () => {
		const div = document.createElement("div");
		ReactDOM.render(<CustomCursor />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("should render correctly", () => {
		render(<CustomCursor />);
		expect(document.querySelector(".cursor > svg > circle")).not.toBeNull();
	});
});
