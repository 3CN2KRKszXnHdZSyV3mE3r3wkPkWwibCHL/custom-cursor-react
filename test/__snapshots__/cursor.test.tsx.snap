// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCursor Component should render correctly 1`] = `
<div
  className="cursor "
>
  <svg
    height={50}
    viewBox="0 0 50 50"
    width={50}
  >
    <circle
      cx={25}
      fill="#000"
      height={25}
      r={12.5}
      stroke="#000"
      strokeWidth={0}
    />
  </svg>
</div>
`;

exports[`it renders correctly 1`] = `
<div
  className="cursor "
>
  <svg
    height={50}
    viewBox="0 0 50 50"
    width={50}
  >
    <circle
      cx={25}
      fill="#000"
      height={25}
      r={12.5}
      stroke="#000"
      strokeWidth={0}
    />
  </svg>
</div>
`;
